[Keycloak CI] Azure and Aurora Migration tests failing

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

The nightly Keycloak CI runs in the last three days failed in migration tests on Aurora and Azure in the `MigrationTest`

The nightly runs are:
**AzureDB:**
Wed 11. 2.: https://github.com/keycloak/keycloak/actions/runs/21893049354/job/63203076981
Thu  12. 2.: https://github.com/keycloak/keycloak/actions/runs/21933813765/job/63343602803
Fri    13. 2.: https://github.com/keycloak/keycloak/actions/runs/21974876771/job/63484347716

**AuroraDB:**
Wed 11. 2.: https://github.com/keycloak/keycloak/actions/runs/21893049354/job/63203076982
Thu  12. 2.: https://github.com/keycloak/keycloak/actions/runs/21933813765/job/63343602804
Fri    13. 2.: _failed because ec2 instance couldn't be created_

```
Error:      "[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 12.90 s <<< FAILURE! -- in org.keycloak.testsuite.migration.MigrationTest",
Error:      "-- [ERROR] org.keycloak.testsuite.migration.MigrationTest -- Time elapsed: 12.85 s <<< ERROR!",
            "-- java.lang.RuntimeException: java.io.IOException: Command build was executed with exit status 1",
            "-- \tat org.keycloak.testsuite.arquillian.containers.KeycloakQuarkusServerDeployableContainer.start(KeycloakQuarkusServerDeployableContainer.java:72)",
            "-- \tat org.jboss.arquillian.container.impl.ContainerImpl.start(ContainerImpl.java:185)",
            "-- \tat org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController$8.perform(ContainerLifecycleController.java:137)",
            "-- \tat org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController$8.perform(ContainerLifecycleController.java:133)",
            "-- \tat org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController.forContainer(ContainerLifecycleController.java:208)",
            "-- \tat org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController.startContainer(ContainerLifecycleController.java:133)",
            "-- \tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)",
            "-- \tat java.base/java.lang.reflect.Method.invoke(Method.java:565)",
            "-- \tat org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)",
            "-- \tat org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)",
            "-- \tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)",
            "-- \tat java.base/java.lang.reflect.Method.invoke(Method.java:565)",
            "-- \tat org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)",
            "-- \tat org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)",
            "-- \tat org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:106)",
            "-- \tat org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)",
            "-- \tat org.keycloak.testsuite.arquillian.AuthServerTestEnricher.startMigratedContainer(AuthServerTestEnricher.java:291)",
            "-- \tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)",
            "-- \tat java.base/java.lang.reflect.Method.invoke(Method.java:565)",
            "-- \tat org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)",
            "-- \tat org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)",
            "-- \tat org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:106)",
            "-- \tat org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)",
            "-- \tat org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:83)",
            "-- \tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)",
            "-- \tat java.base/java.lang.reflect.Method.invoke(Method.java:565)",
            "-- \tat org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)",
            "-- \tat org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)",
            "-- \tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)",
            "-- \tat java.base/java.lang.reflect.Method.invoke(Method.java:565)",
            "-- \tat org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)",
            "-- \tat org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)",
            "-- \tat org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)",
            "-- \tat org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:106)",
            "-- \tat org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeSuite(EventTestRunnerAdaptor.java:71)",
            "-- \tat org.jboss.arquillian.junit.AdaptorManager.initializeAdaptor(AdaptorManager.java:23)",
            "-- \tat org.jboss.arquillian.junit.AdaptorManagerWithNotifier.initializeAdaptor(AdaptorManagerWithNotifier.java:19)",
            "-- \tat org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:103)",
            "-- \tat org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:316)",
            "-- \tat org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:240)",
            "-- \tat org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:214)",
            "-- \tat org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:155)",
            "-- \tat org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)",
            "-- \tat org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)",
            "-- \tat org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)",
            "-- \tat org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)",
            "-- Caused by: java.io.IOException: Command build was executed with exit status 1",
            "-- \tat org.keycloak.testsuite.arquillian.containers.KeycloakQuarkusServerDeployableContainer.executeCommand(KeycloakQuarkusServerDeployableContainer.java:107)",
            "-- \tat org.keycloak.testsuite.arquillian.containers.KeycloakQuarkusServerDeployableContainer.importRealm(KeycloakQuarkusServerDeployableContainer.java:147)",
            "-- \tat org.keycloak.testsuite.arquillian.containers.KeycloakQuarkusServerDeployableContainer.start(KeycloakQuarkusServerDeployableContainer.java:50)",
            "-- \t... 52 more",
            "-- ",
            "-- [INFO] ",
            "-- [INFO] Results:",
            "-- [INFO] ",
Error:      "-- [ERROR] Errors: ",
Error:      "-- [ERROR]   MigrationTest » Runtime java.io.IOException: Command build was executed with exit status 1",
            "-- [INFO] ",
Error:      "-- [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0",
```



### Version

nightly

### Regression

- [ ] The issue is a regression

### Expected behavior

- tests pass

### Actual behavior

- tests fail

### How to Reproduce?

check the recent logs

### Anything else?

_No response_

@vmuzikar - this seems to be caused by the JDK 25 upgrade as there is now the following message: 

> Caused by: java.lang.IllegalArgumentException: Java 25 (69) is not supported by the current version of Byte Buddy which officially supports Java 22 (66) - update Byte Buddy or set net.bytebuddy.experimental as a VM property

--- a/.github/scripts/ansible/roles/keycloak_remote_installer/defaults/main.yml
+++ b/.github/scripts/ansible/roles/keycloak_remote_installer/defaults/main.yml
@@ -3,4 +3,6 @@ kc_home: /opt/keycloak
 m2_home: ~/.m2
 update_system_packages: no
 install_java: yes
-java_version: 25
+java_version:
+  - 25
+  - 21  # TODO: Remove once Keycloak 24 is no longer used for migration tests

--- a/.github/scripts/ansible/roles/keycloak_remote_installer/tasks/install.yml
+++ b/.github/scripts/ansible/roles/keycloak_remote_installer/tasks/install.yml
@@ -10,25 +10,27 @@
     update_cache: yes
 
 
-- name: Install Java {{ java_version }} packages on Ubuntu/Debian
+- name: Install Java {{ item }} packages on Ubuntu/Debian
   when:
     - install_java
     - ansible_os_family == 'Debian'
   package:
     name:
-      - "openjdk-{{ java_version }}-jdk"
-      - "openjdk-{{ java_version }}-jre"
+      - "openjdk-{{ item }}-jdk"
+      - "openjdk-{{ item }}-jre"
     state: present
+  loop: "{{ [java_version] | flatten }}"
 
-- name: Install Java {{ java_version }} packages on RedHat/Amazon
+- name: Install Java {{ item }} packages on RedHat/Amazon
   when:
     - install_java
     - ansible_os_family != 'Debian'
   package:
     name:
-      - "java-{{ java_version }}-openjdk"
-      - "java-{{ java_version }}-openjdk-devel"
+      - "java-{{ item }}-openjdk"
+      - "java-{{ item }}-openjdk-devel"
     state: present
+  loop: "{{ [java_version] | flatten }}"
 
 - name: Install dependencies on the remote hosts
   package: name={{item}} state=present

--- a/.github/scripts/ansible/roles/mvn_remote_runner/tasks/run.yml
+++ b/.github/scripts/ansible/roles/mvn_remote_runner/tasks/run.yml
@@ -16,6 +16,14 @@
   ansible.builtin.shell: |
     set -o pipefail
     cd {{ kc_home }}
+    {% if lookup('env', 'JAVA_VERSION') != "" %}
+    {% if ansible_os_family == 'Debian' %}
+    export JAVA_HOME=/usr/lib/jvm/java-{{ lookup('env', 'JAVA_VERSION') }}-openjdk-amd64
+    {% else %}
+    export JAVA_HOME=/usr/lib/jvm/java-{{ lookup('env', 'JAVA_VERSION') }}-openjdk
+    {% endif %}
+    echo "Using JAVA_HOME: ${JAVA_HOME}"
+    {% endif %}
     echo "{{ mvn_params }}"
     ./mvnw {{ mvn_params }}
   args:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -515,6 +515,7 @@ jobs:
         id: aurora-migration-tests
         env:
           old-version: 24.0.4
+          JAVA_VERSION: 21 # Keycloak 24 is not compatible with Java 25+
         run: |
           EC2_CLUSTER_NAME=${{ steps.ec2-create.outputs.ec2_cluster }}
           AWS_REGION=${{ steps.aurora-init.outputs.region }}
@@ -723,10 +724,11 @@ jobs:
         id: azure-migration-build
         env:
           old-version: 24.0.4
+          JAVA_VERSION: 21 # Keycloak 24 is not compatible with Java 25+
         run: |
           AZ_VM_CLUSTER_NAME=${{ steps.vm-create.outputs.az_vm_cluster }}
           AZURE_REGION=${{ steps.azure-init.outputs.region }}
-          
+
           cd .github/scripts/ansible
           ./mvn_remote_runner.sh ${AZURE_REGION} ${AZ_VM_CLUSTER_NAME} 'clean install -B -DskipTests -pl testsuite/integration-arquillian/servers/migration -Pauth-server-migration -Dmigrated.auth.server.version=${{ env.old-version }} -Dmaven.build.cache.enabled=true -Pdb-mssql-azure -Dkeycloak.connectionsJpa.user=${{ steps.azure-create.outputs.username }} -Dkeycloak.connectionsJpa.password="${{ env.AZURE_DB_PASSWORD }}" -Dkeycloak.connectionsJpa.database=${{ steps.azure-create.outputs.db }} -Dkeycloak.connectionsJpa.url="jdbc:sqlserver://${{ steps.azure-create.outputs.endpoint }}:1433;databaseName=${{ steps.azure-create.outputs.db }};encrypt=true;trustServerCertificate=false;sendStringParametersAsUnicode=false" -Djdbc.driver.tmp.dir=target/unpacked/keycloak-${{ env.old-version }}/providers'
 
@@ -734,6 +736,7 @@ jobs:
         id: azure-migration-tests
         env:
           old-version: 24.0.4
+          JAVA_VERSION: 21 # Keycloak 24 is not compatible with Java 25+
         run: |
           AZ_VM_CLUSTER_NAME=${{ steps.vm-create.outputs.az_vm_cluster }}
           AZURE_REGION=${{ steps.azure-init.outputs.region }}
@@ -749,10 +752,11 @@ jobs:
         id: azure-integration-build
         env:
           old-version: 24.0.4
+          JAVA_VERSION: 21 # Keycloak 24 is not compatible with Java 25+
         run: |
           AZ_VM_CLUSTER_NAME=${{ steps.vm-create.outputs.az_vm_cluster }}
           AZURE_REGION=${{ steps.azure-init.outputs.region }}
-          
+
           cd .github/scripts/ansible
           ./mvn_remote_runner.sh ${AZURE_REGION} ${AZ_VM_CLUSTER_NAME} 'clean install -B -DskipTests -pl testsuite/integration-arquillian/servers/migration -Pauth-server-migration -Dmigrated.auth.server.version=${{ env.old-version }} -Dmaven.build.cache.enabled=true -Pdb-mssql-azure -Dkeycloak.connectionsJpa.user=${{ steps.azure-create.outputs.username }} -Dkeycloak.connectionsJpa.password="${{ env.AZURE_DB_PASSWORD }}" -Dkeycloak.connectionsJpa.database=${{ steps.azure-create.outputs.db }} -Dkeycloak.connectionsJpa.url="jdbc:sqlserver://${{ steps.azure-create.outputs.endpoint }}:1433;databaseName=${{ steps.azure-create.outputs.db }};encrypt=true;trustServerCertificate=false;sendStringParametersAsUnicode=false" -Djdbc.driver.tmp.dir=target/unpacked/keycloak-${{ env.old-version }}/providers'
 

